File: src/main/resources/public/ts/services/setting.service.ts
import {ng} from 'entcore';
import http, {AxiosResponse} from 'axios';
import {ISettingResult, Setting} from "../models/setting.model";

export interface ISettingService {
    getGlobalSettings(): Promise<Setting>;
}

export const settingService: ISettingService = {
    /**
     * Get global settings.
     */
    getGlobalSettings: async (): Promise<Setting> =>
        http.get(`/minibadge/global-settings`)
            .then((res: AxiosResponse) => new Setting(<ISettingResult>res.data))
};

export const SettingService = ng.service('SettingService', (): ISettingService => settingService);

File: src/main/java/fr/openent/minibadge/service/SettingService.java
package fr.openent.minibadge.service;

import fr.openent.minibadge.model.GlobalSettings;
import io.vertx.core.Future;
import org.entcore.common.user.UserInfos;

public interface SettingService {

    /**
     * get global settings
     * @return global settings
     */
    Future<GlobalSettings> getGlobalSettings(UserInfos user);
}
